commit d1495bdb008905ee51518ef7b7891f99311c0a8c (refs/changes/51/3351/1)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2018-02-09T18:21:19-08:00 (1 year, 8 months ago)
    
    TRITON-129 workflow job task throws uncaught exception in some error cases when provisioning volumes

diff --git a/lib/workflows/nfs-volumes/index.js b/lib/workflows/nfs-volumes/index.js
index f443c65..d1b65ef 100644
--- a/lib/workflows/nfs-volumes/index.js
+++ b/lib/workflows/nfs-volumes/index.js
@@ -17,9 +17,7 @@
  */
 
 var vasync = require('vasync');
-var VError = require('verror');
-
-var common = require('../job-common');
+var verror = require('verror');
 
 function provisionNfsVolumes(job, cb) {
     var requiredVolumes;
@@ -77,7 +75,7 @@ function provisionNfsVolumes(job, cb) {
             job.createdVolumes = createdVolumes;
             cb(null, 'All volumes provisioned');
         } else {
-            cb(new VError(err, 'Could not provision volumes'));
+            cb(new verror(err, 'Could not provision volumes'));
         }
     });
 
@@ -334,8 +332,6 @@ function provisionNfsVolumes(job, cb) {
                         next(volumeCreationErr);
                         return;
                     }
-
-                    next(volumeCreationErr);
                 }
             }]}, function onVolumeProvDone(volProvErr) {
                 if (provisionContext.volumeReservation === true &&
